fog/sql_recovery_db/src/lib.rs

```diff
@@ -2655,6 +2679,90 @@ mod tests {
         let db_test_context = test_utils::SqlRecoveryDbTestContext::new(logger);
         let db = db_test_context.get_db_instance();
 
+        // At first, there are no records.
+        assert_eq!(
+            db.get_ingress_key_records(
+                0,
+                IngressPublicKeyRecordFilters {
+                    should_include_lost_keys: false,
+                    should_include_retired_keys: false,
+                    should_only_include_unexpired_keys: false,
+                }
+            )
+            .unwrap(),
+            vec![],
+        );
+
+        // Add an ingress key and see that we can retreive it.
+        let ingress_key1 = CompressedRistrettoPublic::from_random(&mut rng);
+        db.new_ingress_key(&ingress_key1, 0).unwrap();
+
+        let ingress_key2 = CompressedRistrettoPublic::from(RistrettoPublic::from_random(&mut rng));
+        db.new_ingress_key(&ingress_key2, 5).unwrap();
+
+        // Add an ingest invocation and see that we can see it.
+        let invoc_id1 = db
+            .new_ingest_invocation(None, &ingress_key1, &random_kex_rng_pubkey(&mut rng), 123)
+            .unwrap();
+
+        let ranges = db.get_ingestable_ranges().unwrap();
+        assert_eq!(ranges.len(), 1);
+        assert_eq!(ranges[0].start_block, 123);
+        assert_eq!(ranges[0].decommissioned, false);
+        assert_eq!(ranges[0].last_ingested_block, None);
+
+        // This makes last_scanned_block equal 10.
+        for block_index in 0..11 {
+            let (block, records) = random_block(&mut rng, block_index, 10);
+            db.add_block_data(&invoc_id1, &block, 0, &records).unwrap();
+        }
+
+        db.set_report(
+            &ingress_key1,
+            "",
+            &ReportData {
+                pubkey_expiry: 20,
+                ingest_invocation_id: None,
+                report: Default::default(),
+            },
+        )
+        .unwrap();
+
+        db.retire_ingress_key(&ingress_key1, true).unwrap();
+        db.report_lost_ingress_key(ingress_key2).unwrap();
+
+        let actual = db
+            .get_ingress_key_records(
+                0,
+                IngressPublicKeyRecordFilters {
+                    should_include_lost_keys: false,
+                    should_include_retired_keys: true,
+                    should_only_include_unexpired_keys: true,
+                },
+            )
+            .unwrap();
+
+        let expected = IngressPublicKeyRecord {
+            key: ingress_key1.clone(),
+            status: IngressPublicKeyStatus {
+                start_block: 0,
+                pubkey_expiry: 20,
+                retired: true,
+                lost: false,
+            },
+            last_scanned_block: Some(10),
+        };
+        assert_eq!(actual[0], expected);
```

Comment:
I suggest making the test more strict and comparing `actual` to `vec![expected]`.

Reply:
Done.